### DISCUSSION TITLE
Ignoring unrasiseable from offending libraries

### DISCUSSION THREAD

**Discussion by awichmann-mintel** [pytest-dev/pytest#8340] [Q&A]
Hello,
[The new unraiseable exception feature in pytest 6.2](https://docs.pytest.org/en/stable/usage.html#warning-about-unraisable-exceptions-and-unhandled-thread-exceptions) is throwing errors for unclosed resources in botocore. [Since this is an ongoing issue](https://github.com/boto/boto3/issues/454#issuecomment-586033745), my team and I would like to ignore these warnings only from certain libraries.

Our best solution is to filter all ResourceWarnings from `_pytest` with the `warnings` module in a fixture like so...
```python
@pytest.fixture(scope="session", autouse=True)
def silence_resource_warnings():
    warnings.filterwarnings(
        action="ignore", message="unclosed", category=ResourceWarning, module="_pytest"
    )
```

Is this advisable? Is there a way to target the module that is actually not cleaning up these resources?

edit: additionally, if this were possible, do you have any advice on how to test such a fixture silences the correct libraries?

**Comment by bluetech** [comment]
If by "certain libraries" here you mean pytest rather than boto3, then you can ignore it by adding a `ignore::pytest.PytestUnraisableExceptionWarning` warning filter (pytest has an [option](https://docs.pytest.org/en/stable/reference.html) for this, you don't need to use a session fixture), or by disabling the plugin by passing the `-p no:unraisableexception` option to pytest.

If you mean boto3, there is no way to reliably filter ResourceWarnings from a specific module AFAIK.

**Reply by Andrew-Wichmann** [reply]
> If you mean boto3, there is no way to reliably filter ResourceWarnings from a specific module AFAIK.

yeah, that's what I'm going for =/ oh well
